docs/reference/commandline/pull.md

```diff
@@ -24,31 +24,195 @@ Most of your images will be created on top of a base image from the
+use `docker pull`.
+
+## Examples
+
+### Pull an image from Docker Hub
+
+To download a particular image, or set of images (i.e., a repository), use
+`docker pull`. If no tag is provided, Docker uses the `:latest` tag as a default.
+This command pulls the `debian:latest` image:
+
+```bash
+$ docker pull debian
+
+Using default tag: latest
+latest: Pulling from library/debian
+fdd5d7827f33: Pull complete
+a3ed95caeb02: Pull complete
+Digest: sha256:e7d38b3517548a1c71e41bffe9c8ae6d6d29546ce46bf62159837aad072c90aa
+Status: Downloaded newer image for debian:latest
+```
+
+Docker images can consist of multiple layers. In the example above, the image
+consists of two layers; `fdd5d7827f33` and `a3ed95caeb02`.
+
+Layers can be reused by images. For example, the `debian:jessie` image shares
+both layers with `debian:latest`. Pulling the `debian:jessie` image will therefore
+only pull its meta-data, but not its layers, because all layers are already
+present locally:
+
+```bash
+$ docker pull debian:jessie
+
+jessie: Pulling from library/debian
+fdd5d7827f33: Already exists
+a3ed95caeb02: Already exists
+Digest: sha256:a9c958be96d7d40df920e7041608f2f017af81800ca5ad23e327bc402626b58e
+Status: Downloaded newer image for debian:jessie
+```
+
+To see which images are present locally, use the [`docker images`](images.md)
+command:
+
+```bash
+$ docker images
+
+REPOSITORY   TAG      IMAGE ID        CREATED      SIZE
+debian       jessie   f50f9524513f    5 days ago   125.1 MB
+debian       latest   f50f9524513f    5 days ago   125.1 MB
+```
+
+Docker uses a content-addressable image store, and the image ID is a sha256-digest
+of the image layers. In the example above, `debian:jessie` and `debian:latest`
+have the same image ID because they are actually the *same* image tagged with
+different names. Because they are the same image, their layers are stored only
+once and do not consume extra disk space. 
+
+For more information about images, layers, and the content-addressable store,
+refer to [understand images, containers, and storage drivers](../../userguide/storagedriver/imagesandcontainers.md).
+
+
+## Pull an image by digest (immutable tag)
+
+So far, you've pulled images by their name (and "tag"). Using names and tags is
+a convenient way to work with images. When using tags, you can `docker pull` an
+image again to make sure you have the most up-to-date version of that image.
+For example, `docker pull ubuntu:14.04` pulls the latest version of the Ubuntu
+14.04 image.
+
+In some cases, you don't want images to be updated to newer versions, but
+prefer to use a fixed version of an image. Docker enables you to pull an image
+by its *digest*. When pulling an image by digest, you're specifying *exactly*
+which version of an image to pull. Doing so, allows you to "pin" an image to a
+specific version, and guaruantee that the image you're using is always the same.
+
+To know the digest of an image, pull the image first. Let's pull the latest
```

Comment:
Yes, I decided to keep it simple for now, but perhaps we can add something at a later stage.